src/MentionsInput.js

```diff
@@ -280,6 +319,143 @@ class MentionsInput extends React.Component {
     }
   }
 
+  handlePaste(event) {
+    if (event.target !== this.inputRef) {
+      return
+    }
+
+    event.preventDefault()
+
+    const { selectionStart, selectionEnd } = this.state
+    const { value, children } = this.props
+
+    const config = readConfigFromChildren(children)
+
+    const realStartIndex = mapPlainTextIndex(
+      value,
+      config,
+      selectionStart,
+      'START'
+    )
+    const realEndIndex = mapPlainTextIndex(value, config, selectionEnd, 'END')
+
+    const pastedMentions = event.clipboardData.getData('text/react-mentions')
+    const pastedData = event.clipboardData.getData('text/plain')
+
+    const newValue = spliceString(
+      value,
+      realStartIndex,
+      realEndIndex,
+      pastedMentions || pastedData
+    )
+    const newPlainTextValue = getPlainText(newValue, config)
+
+    const eventMock = { target: { ...event.target, value: newValue } }
+
+    this.executeOnChange(
+      eventMock,
+      newValue,
+      newPlainTextValue,
+      getMentions(newValue, config)
+    )
+  }
+
+  handleCopy(event) {
+    if (event.target !== this.inputRef) {
+      return
+    }
+
+    const { selectionStart, selectionEnd } = this.state
+    const { children, value } = this.props
+
+    const config = readConfigFromChildren(children)
+
+    const realStartIndex = mapPlainTextIndex(
```

Comment:
how about `markupStartIndex` / `markupEndIndex`? Not sure if one value is more "real" than the other...

Reply:
Absolutely right.